### ISSUE TITLE
WindowsFormsIntegration Compatability issue.

### ISSUE THREAD

**Issue by AaronAnz** [unosquare/ffmediaelement#311]
# WindowsFormsIntegration Compatability issue

Compatability issues with WindowsFormsIntegration and possibly .net framework 4.7.2

## Issue Categories

 - [x] Bug
 - [ ] Feature Request
 - [ ] Question
 - [ ] Not sure

## Version Information
I tried both the checked options in two seperate instances.
 - [x] NuGet Package 4.0.270
 - [x] Build From *Master* branch, Commit 3c3733c 
 - [ ] Build from *Branch Name*, Commit N/A

## Steps to Reproduce
* I'm using 4.7.2 but it seems to not be the cause because commit 3c3733c runs fine on 4.7.2 when not using WindowsFormsIntegration *
* For the sake of testing I did this within the same solution as commit 3c3733c *
1. Create WPF UserControl Library
2. Add the following references:
            1. Unosquare.FFME.Common
            2. Unosquare.FFME.Windows
3. Add an FFME control to the UserControl that was generated.
4. Point Unosquare.FFME to path of all ffmpeg exes and dlls. 
5. Build the control library.
6. Create WinForm application.
7. Add the following assemblies as references:
            * Adding Unosquare.FFME references to this application doesn't seem to change this behavior *
            1. PresentationCore
            2. PresentationFramework
            3. WindowsBase
            4. WindowsFormIntegration
            5. The WPF UserControl Library you created earlier
8. Build the solution to be sure everything is good. 
9. Try to drag the WPF UserControl from the Toolbox into the WinForm you created and you will receive an error like [these errors](https://i.imgur.com/kBS7fqa.png)

## Expected Results
Message of: 
- "Failed to create component ..."

with details of: 
 -  "System.TypeInitializationException"
 or 
- "XamlParseException: SetConnectionId threw an exception"

[errors](https://i.imgur.com/kBS7fqa.png)
 
## Sample Code

[ffmediaelement-master-with-sample-bug.zip](https://github.com/unosquare/ffmediaelement/files/2775133/ffmediaelement-master-with-sample-bug.zip)
* Build the solution and then go to the WinForms project and try to drag either WPF UserControl onto the form. *

**Comment by AaronAnz** [comment]
after I experienced the aforementioned, I decided to try other libraries. None of them worked well so I ended up back here. I took a peek under the hood and here is the issue.
https://github.com/unosquare/ffmediaelement/blob/3c3733cad72a8d70cb2220ac44cfd5314bf073de/Unosquare.FFME.Common/Shared/Constants.cs#L20 

Remove lines 20 and 21, and change 26 to this and it fixed it. Not really sure why WindowsFormsIntegration hates that part.

`public static string FFmpegSearchPath => Path.GetFullPath(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location) ?? ".");`

I'm glad I get to use this project now.

**Comment by mariodivece** [comment]
I know why this occurs. I am reopening the issue

**Comment by mariodivece** [comment]
I will fix when I get some spare time

**Comment by AaronAnz** [comment]
Yeah no worries, I got it working in the meantime. Glad I get to use your project. :)

**Comment by mariodivece** [comment]
I have changed the initialization code. Could you try again with the master branch? Also, please note I have bumped up the version to FFmpeg 4.1

**Comment by AaronAnz** [comment]
I tested commit 9154e7f but this time I left it at 4.6.1. I built everything then tried to drag a user control that contained the ffmediaelement onto the WinForm and it gave me a similar issue to last time but not quite the same ( I closed out before I could screencap it and it didnt occur again ). I rebuilt the project closed the solution and reopened and seems to be working OK from then on. I think that was an isolated incident as I can't reproduce it with this solution and don't really want to build another to test.

tl;dr: seems to be working, had an error but I think it was an isolated incident.
